src/components/Search/Search.tsx

```diff
@@ -53,30 +51,39 @@ export interface SearchProps extends React.HTMLAttributes<HTMLDivElement> {
 }
 
 const Search = ({
-  useIcon = true,
   onSearch,
   searchResults = [],
   state = 'idle',
   defaultValue = '',
   className,
-  placeholder = 'Type to search',
+  placeholder,
   searchButtonText = 'Search',
-  loadingText = 'Loading...',
+  loadingText = 'Search In Progress...',
   invalidText = 'Invalid search',
   getAllResultsText = (searchValue) => `View all results for ${searchValue}`,
   getAllResultsLink = (searchValue) => `/Search?Search=${searchValue}`,
   ...props
 }: React.PropsWithChildren<SearchProps>) => {
   const { className: baseClassName, 'data-testid': baseTestId, ...commonProps } = getCommonProps(props, 'Search');
-  const [overlayEnabled, setOverlayEnabled] = React.useState(false);
+  const [searchEnabled, setSearchEnabled] = React.useState(false);
   const searchInputRef = React.useRef<HTMLInputElement>(null);
-
+  const searchFormRef = React.useRef<HTMLFormElement>(null);
   const value = searchInputRef.current?.value;
+  const renderSpinner = state === 'loading' || state === 'submitting';
   const showSearch = () => {
-    setOverlayEnabled(!overlayEnabled);
+    setSearchEnabled(!searchEnabled);
     // means we're opening search
-    if (!overlayEnabled) {
-      searchInputRef.current?.focus();
+    if (searchEnabled) {
+      searchFormRef.current?.reset();
+      return;
+    }
+    searchInputRef.current?.focus();
+  };
+
+  const onKeyDown = (e: React.KeyboardEvent<HTMLInputElement> | React.KeyboardEvent<HTMLAnchorElement>) => {
+    if (e.key === 'Escape') {
+      setSearchEnabled(false);
+      searchFormRef.current?.reset();
     }
```

Comment:
when I hit `Enter` in the search input it doesn't seem to take me to the `Search?Search=value` like cliking the View All Details link does